### PR TITLE
fillna function rewrite

### PR DESCRIPTION
**Pull request by RainFung**
create a new PR base on https://github.com/databricks/koalas/pull/576

### CODE REVIEW

**Comment by RainFung** [comment]
@ueshin `If the integer series doesn't have null values, it preserves the integer type.`
```python
>>> pd.Series([1,2,3,4]).fillna(method='ffill')
0    1
1    2
2    3
3    4
dtype: int64
```

Can you help me solve this problem?

**Comment by HyukjinKwon** [comment]
cc @ueshin

File: databricks/koalas/series.py [hunk 1]
```diff
@@ -1210,8 +1227,72 @@ def fillna(self, value=None, axis=None, inplace=False):
...
+                    end = Window.unboundedFollowing
+            else:
+                raise ValueError('Expecting pad, ffill, backfill or bfill.')
+            window = Window.orderBy(self._internal.index_scols).rowsBetween(begin, end)
+            scol = F.when(scol.isNull(), func(scol, True).over(window)).otherwise(scol)
+
+        kseries = Series(self._kdf._internal.copy(scol=scol), anchor=self._kdf)\
+            .rename(column_name)
+        if isinstance(spark_type, NumericType):
+            kseries = kseries.astype(np.float64)
```
Comment (ueshin) [inline databricks/koalas/series.py]: I'm wondering why we need this cast?

File: databricks/koalas/series.py [hunk 2]
```diff
@@ -1210,8 +1227,72 @@ def fillna(self, value=None, axis=None, inplace=False):
...
+        3       c
+        4       c
+        Name: x, dtype: object
         """
-        kseries = _col(self.to_dataframe().fillna(value=value, axis=axis, inplace=False))
+        if self.isnull().sum() == 0:
+            if inplace:
+                return
+            else:
+                return self
```
Comment (ueshin) [inline databricks/koalas/series.py]: Maybe we should return a copy to avoid the effect of the following inplace-like operations?

**Comment by RainFung** [comment]
Fixed

**Pull request merged**